src/DBMLib/lib_dbm.rs

```diff
@@ -647,17 +648,40 @@ pub fn rs_dbm_fed_minus_fed(
             &mut res,
         );
 
-        let result = rs_fed_to_vec(&mut res);
+        return fed_to_federation(&mut res, dim);
+    }
+}
 
-        return result;
+fn fed_to_federation(fed: &mut dbm_fed_t, dim: u32) -> Federation {
+    let result = rs_fed_to_vec(fed);
+
+    // zone: [i32; dim * dim]
+    let mut zones = Vec::with_capacity(result.len());
+    for dbm_ptr in result.iter() {
+        if *dbm_ptr == std::ptr::null() {
+            continue;
+        }
```

Comment:
Den tidligere kode filterede ikke null pointers fra gjorde den, så jeg ved ikke om vi skal gøre det? Da en null ptr har en effekt på refinement checket, men det er måske en fejl.

Så det skal vi lige overveje før vi merger koden.

Reply:
I think it's fine to keep it, we do filter in master in the input_enable check: https://github.com/Ecdar/Reveaal/blob/967549d00faec2fbd568fac5f90c3fad44f5ede9/src/System/input_enabler.rs#L105 
so we can avoid that by checking here. Generally i think it's good that the zones in a federation are never null if we ever iterate over them elsewhere